### ISSUE TITLE
Change task output to changed when results.changed is true

### ISSUE THREAD

**Issue by caljess599** [ansible/ansible#33640]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
template

##### ANSIBLE VERSION
```
ansible 2.4.2.0 (stable-2.4 b6c117ebdd) 
```
##### CONFIGURATION

DEFAULT_GATHER_SUBSET  =  hardware
DEFAULT_REMOTE_USER = root

##### OS / ENVIRONMENT
Ubuntu 16.04

##### SUMMARY
(This is more of a complaint than a feature idea, since I could swear Ansible used to work this way, and so the Rubicon has likely been crossed in this regard. But today I find myself really bummed about it.)

Here goes:

How does Ansible determine when to report a task as changed rather than ok in task output? I simply don't understand why this decision is not governed by the results.changed return value, which is the value used to calculate in the summary how many tasks resulted in changes. When this return value is 'true', the task should report as CHANGED, not OK. Otherwise, in a given run, you cannot tell WHICH task(s) have changed--unless you add a debug statement to each task, which is a lot of work! Frankly, I'm not sure anymore *how* Ansible determines when to report a task as CHANGED if results.changed is true doesn't do it. 
 
##### STEPS TO REPRODUCE
```yaml
- name: for example, the template task
  template:
    dest: /home/in/dirfoo
    src: thisfile.j2  
  register: results

- debug: var=results.changed
```
##### EXPECTED RESULTS
In the case where the template contents or attributes have changed...
```
TASK [role : play.yml | for example, the template task] *****************
changed: [my-host]

TASK [role : play.yml | debug ] *****************
ok: [my-host] => {
    "results.changed": true

PLAY RECAP *****************************************************************************
my-host                   : ok=2   changed=1    unreachable=0    failed=0   
```

##### ACTUAL RESULTS
In the case where the template contents or attributes have changed...
```
TASK [role : play.yml | for example, the template task] *****************
ok: [my-host]

TASK [role : play.yml | debug ] *****************
ok: [my-host] => {
    "results.changed": true

PLAY RECAP *****************************************************************************
my-host                   : ok=2   changed=1    unreachable=0    failed=0   
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/notification/hall.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/notification/hall.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @bkimble
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/files/template.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/files/template.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
I cannot reproduce the behavior that you are describing.  If the file contents or attributes change, and `results.changed` is `true` it always shows as being changed.

Can you run your playbook with `-vvv` and provide the output here?

needs_info

**Comment by caljess599** [comment]
I'd be happy to hear that I'm doing something wrong and that this behavior is not expected. Here's some verbose output (a bit manipulated for company privacy) of the template step in action.

[ansible_verbose.txt](https://github.com/ansible/ansible/files/1536319/ansible_verbose.txt)

**Comment by sivel** [comment]
I'm not sure what is going on yet.  I can see from the module return data that `changed: true` is there, but the host output doesn't contain it:

```
ok: [my-host] => {
    "checksum": "3ef988d8e23215a827b3fda1d4910d0f0cb785d6",▒
    "dest": "/home/webapp/.bash_fu.d/passenger",▒
    "diff": [],▒
    "gid": 1001,▒
    "group": "webapp",▒
    "invocation": {
```

From my test:

```
changed: [localhost] => {
    "changed": true,
    "checksum": "53a436819cbf90f544a3e76709b5c62cecd96140",
    "dest": "./thisfile.out",
    "diff": [],
    "gid": 20,
    "group": "staff",
    "invocation": {
```

And somehow, debug can still see it?

Just making sure, you didn't accidentally delete that `changed: true` line when cleaning the data correct?

**Comment by caljess599** [comment]
Doesn't look like a typo... sadly. It is pretty weird. `changed: true` is there in the verbose output printed before the task report (which maybe explains why results can grab it and show its status as changed?):

`<my-host> (0, '\r\n\r\n{"src": "/home/thisuser/.ansible/tmp/ansible-tmp-1512582856.2-5770613196913/source",` "**changed": true**, `"group": "webapp", "uid": 1001, "dest": "/home/webapp/.bash_fu.d/passenger", "checksum": "3ef988d8e23215a827b3fda1d4910d0f0cb785d6", "md5sum": "d689ea167ab09e6c95643ea62f5e3221", "owner": "webapp", "state": "file", "gid": 1001, "mode": "0664", "invocation": {"module_args": {"directory_mode": null, "force": true, "remote_src": null, "owner": "webapp", "follow": false, "local_follow": null, "group": "webapp", "unsafe_writes": null, "setype": null, "content": null, "serole": null, "dest": "/home/webapp/.bash_fu.d/passenger", "selevel": null, "original_basename": "passenger.j2", "regexp": null, "validate": null, "src": "/home/thisuser/.ansible/tmp/ansible-tmp-1512582856.2-5770613196913/source", "seuser": null, "delimiter": null, "mode": "0664", "attributes": null, "backup": false}}, "size": 1543}\r\n', 'Shared connection to my-host closed.\r\n')`

But there is no `changed: true` line in the subsequent task report output--and the task is marked as `ok`!

```
ok: [my-host] => {
    "checksum": "3ef988d8e23215a827b3fda1d4910d0f0cb785d6", 
    "dest": "/home/webapp/.bash_fu.d/passenger", 
    "diff": [], 
    "gid": 1001, 
    "group": "webapp", 
    "invocation": {
        "module_args": {
            "attributes": null, 
            "backup": false, 
            "content": null, 
            "delimiter": null, 
            "dest": "/home/webapp/.bash_fu.d/passenger", 
            "directory_mode": null, 
            "follow": false, 
            "force": true, 
            "group": "webapp", 
            "local_follow": null, 
            "mode": "0664", 
            "original_basename": "passenger.j2", 
            "owner": "webapp", 
            "regexp": null, 
            "remote_src": null, 
            "selevel": null, 
            "serole": null, 
            "setype": null, 
            "seuser": null, 
            "src": "/home/thisuser/.ansible/tmp/ansible-tmp-1512582856.2-5770613196913/source", 
            "unsafe_writes": null, 
            "validate": null
        }
    }, 
    "md5sum": "d689ea167ab09e6c95643ea62f5e3221", 
    "mode": "0664", 
    "owner": "webapp", 
    "size": 1543, 
    "src": "/home/thisuser/.ansible/tmp/ansible-tmp-1512582856.2-5770613196913/source", 
    "state": "file", 
    "uid": 1001
}
```

**Comment by sivel** [comment]
@caljess599 Can you run `ansible-playbook` with `-vvvv` (4) and provide all lines that look like:

```
Loading callback plugin default of type stdout...
```

I'd like to know if there are multiple callbacks loaded and what they are.

needs_info

**Comment by caljess599** [comment]
There's just this one:

```
Loading callback plugin default of type stdout, v2.0 from /my/home/path/lib/ansible/plugins/callback/__init__.pyc
```

**Comment by bcoca** [comment]
that is wrong, default is in default.py or default.pyc

**Comment by sivel** [comment]
@bcoca It might be a bug in how that is displayed, but mine shows `__init__.py` for that line as well.

Still doesn't explain what is happening to the `changed` key.

**Comment by bcoca** [comment]
yep, fixed it, should show correct file now

**Comment by caljess599** [comment]
Okay, so it occurred to me that this behavior probably started when I switched to using the latest apt release I'd installed to following the git code repo. I uninstalled that release, did a git pull (for the heck of it), and voila, it works as expected! 

FWIW, that line is still `Loading callback plugin default of type stdout, v2.0 from /path/to/ansible/lib/ansible/plugins/callback/__init__.py` 

Thank you so much for helping me with this... It was all my fault so I appreciate the assistance in getting it resolved.
